src/transformers/modeling_utils.py

```diff
@@ -658,6 +658,11 @@ def enforce_repetition_penalty_(self, lprobs, batch_size, num_beams, prev_output
     def generate(
         self,
         input_ids=None,
+        encoder_input_ids=None,
+        decoder_input_ids=None,
+        attention_mask=None,
+        encoder_attention_mask=None,
+        decoder_attention_mask=None,
```

Comment:
1) what if attention_mask and enc_attention_mask both specified?
2) when would you need decoder_attention_mask?

Reply:
The naming is confusing because this decoder_attention_mask does not correspond to argument names for seq2seq models. I had to read the code to figure out what was going on.

Reply:
I was confused about the uses of attention-mask, hadn't realized that seq2seq models only take encoder-side attention. Will switch that back to a single attention_mask value in the next commit.